### PR TITLE
:sparkles: New probe for required MFA

### PR DESCRIPTION
**Pull request by eddie-knight**
#### What kind of change does this PR introduce?

New Probe to check whether an organization has MFA enabled. Requires an authorized token. Only supports GitHub in the present state.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

NONE

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Added independent probe to validates that the organization requires MFA for all collaborators.
```

### CODE REVIEW

**Comment by eddie-knight** [comment]
Note that this follows #4391

File: clients/githubrepo/client.go [hunk 1]
```diff
@@ -204,6 +204,17 @@ func (client *Client) GetCreatedAt() (time.Time, error) {
 	return client.repo.CreatedAt.Time, nil
 }
 
+func (client *Client) GetMFARequired() (required bool, err error) {
+	org, _, err := client.repoClient.Organizations.Get(context.Background(), client.repourl.owner)
```
Comment (raghavkaul) [inline clients/githubrepo/client.go:208]: Could we add e2e tests of this on:
1. repo owned by a user
2. repo owned by an organization

File: clients/gitlabrepo/client.go [hunk 2]
```diff
@@ -236,6 +236,11 @@ func (client *Client) GetCreatedAt() (time.Time, error) {
 	return client.project.getCreatedAt()
 }
 
+func (c *Client) GetMFARequired() (required bool, err error) {
```
Comment (raghavkaul) [inline clients/gitlabrepo/client.go:239]: Appears this is available on GitLab for group namespaces as well:
https://docs.gitlab.com/ee/api/groups.html

File: probes/orgRequiresMFA/def.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: orgRequiresMFA # required
+lifecycle: experimental # required
+short: A short description of this probe
+motivation: >
+ What is the motivation for this probe?
+implementation: >
+ How does this probe work under-the-hood?
```
Comment (raghavkaul) [inline probes/orgRequiresMFA/def.yml:21]: Could you add these fields to the documentation?

File: probes/orgRequiresMFA/def.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+outcome:
+ - If MFA is found to be required, the probe returns OutcomeTrue
+ - IF MFA is not found to be required, the probe returns OutcomeFalse
+ - If the runtime environment does not have authentication for the target project, the probe returns OutcomeNotAvailable
```
Comment (raghavkaul) [inline probes/orgRequiresMFA/def.yml:25]:
```suggestion
 - If MFA is required by the organization that owns the repo, the probe returns OutcomeTrue
 - IF MFA is not required by the organization that owns the repo, the probe returns OutcomeFalse
 - If the runtime environment does not have authentication for the target project, the probe returns OutcomeNotAvailable
```

File: clients/repo_client.go [hunk 5]
```diff
@@ -44,6 +44,7 @@ type RepoClient interface {
 	GetCreatedAt() (time.Time, error)
 	GetDefaultBranchName() (string, error)
 	GetDefaultBranch() (*BranchRef, error)
+	GetMFARequired() (bool, error)
```
Comment (spencerschrock) [inline clients/repo_client.go:47]: We may want to reach a decision on https://github.com/ossf/scorecard/issues/4049

This is technically a breaking change.

**Pull request closed without merging**